[DOC-2194] docs(RestAPI): create doc for gsql builtin endpoints;

ArthurXie-tg opened the pull request.

zhihuapengtg committed:
init

zhihuapengtg committed:
Update index.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

update rest api for edges

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

schema change

zhihuapengtg committed:
Update gsql-endpoints.adoc

update schema/data-sources/user/group/ etc rest api documents

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
refine

quanchentg committed:
make pr

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+
+`POST /gsql/v1/schema/vertices`
+
+This endpoint is used to create global vertices using gsql command
```
Need to specify we accept plain text as the payload, and the payload is the gsql command

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+
+| `gsql`
+| yes
+| this should be false, means a json format
```
Not clear, should be:

`False means payload would be json format, True means payload would be query string, which is the gsql command to add global vertex to local graph.`

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+
+| `vertexName`
+| yes
+| the vertex types to be delete, if there exist more than one, use , separate it. value all means dorp all the vertices
```
Barely readable, please ask ChatGPT to refine.

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+
+| `gsql`
+| yes
+| this should be false, means a json format
```
ditto

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+
+=== drop local vertices
+
+`DELETE /gsql/v1/schema/vertices`
```
What's the difference between this one and `drop global vertices`? If the API are the same, we'd better combine them into one.

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+'''  
+=== drop a global vertex
+
+`DELETE /gsql/v1/schema/vertices/{vertexName}`
```
Combine `drop a global vertex` and `drop a local vertex`

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+'''  
+=== update a global vertex attribute
+
+`PUT /gsql/v1/schema/vertices/{vertexName}`
```
Ditto and same for the following part, combine them if their API are the same. This page is huge, we probably do not want to make it even more complicated.

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+===  show all indexes
+`GET /gsql/v1/schema/indexes`
+
+This endpoint is used to show all indexes on a vertex.
```
We should explain what does index means here for user.

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+
+| `vertex`
+| yes
+| the vertex on which to drop index
```
`which vertex to drop index on`

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+--
+[source,bash]
+----
+curl -X DELETE -H "content-type: text/plain" "http://localhost:14240/gsql/v1/schema/indexes?graph=g&vertex=name1 &indexName=index1,index2"
```
Is the space between `vertex=name1` and `&indexName=index1,index2` intentional?

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,4415 @@
+--
+[source,bash]
+----
+curl -X POST -H "content-type: application/json" "http://localhost:14240/gsql/v1/schema/vertices?gsql=false" -d '{"createVertices":[{"Config":{"STATS":"OUTDEGREE_BY_EDGETYPE"},"Attributes":[{"AttributeType":{"Name":"STRING"},"AttributeName":"name"}],"PrimaryId":{"AttributeType":{"Name":"UINT"},"AttributeName":"user_id"},"Name":"User5"},
+{"Config":{"STATS":"OUTDEGREE_BY_EDGETYPE"},"Attributes":[{"AttributeType":{"Name":"STRING"},"AttributeName":"name"}],"PrimaryId":{"AttributeType":{"Name":"UINT"},"AttributeName":"user_id"},"Name":"User4"}
+]}'
```
It's also good to put complex sample request in multiple lines, but we need to refactor it to better format

zhihuapengtg committed:
Update gsql-endpoints.adoc

address comments

quanchentg committed:
REST API DRAFT

quanchentg committed:
REST API DRAFT

zhihuapengtg committed:
Update gsql-endpoints.adoc

address comments

ArthurXie-tg committed:
Update gsql-endpoints.adoc

quanchentg committed:
REST API Doc Refactor

quanchentg committed:
add parameters

quanchentg committed:
refactor

quanchentg committed:
comment addressed

quanchentg committed:
Merge pull request #160 from tigergraph/quan/DOC-2194

rest api quan&xun

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+--
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X GET 'localhost:8123/gsql/v1/saml/meta' 
```
@quanchentg Wrong example, this is for oidc login

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+| `clear`
+| no
+| default:false. clear is true means clear out the current block list.
```
@quanchentg  `true means clear out the current block list, default value is false.`

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+==== Example
+[,tabs]
+====
+Request::
```
@zhihuapengtg Change to `Sample Request::`, same for below

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+curl -X GET -H "content-type: text/plain" "http://localhost:14240/gsql/v1/schema/indexes?graph=recommend" 
+----
+--
+Response::
```
@zhihuapengtg  Change to `Sample Response::`, same for below

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+`GET /gsql/v1/schema/edges/{edgeName}`
+
+This endpoint is used to retrieve information about a specific edge type within a local graph or retrieve information about a specific global edge type.
```
@zhihuapengtg Please add a comma:
```
This endpoint is used to retrieve information about a specific edge type within a local graph, or retrieve information about a specific global edge type.
```

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+| `graph`
+| yes
+| Specifies the name of the graph for which the schema details are to be shown.
```
@zhihuapengtg `Specifies for which graph the schema details to be shown.`

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+| `graph`
+| yes
+| the graph who the local schema change runs on, If not provided, means running global schema change job.
```
@zhihuapengtg  `which graph to run the local schema change job on, ...`

ArthurXie-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+| `graph`
+| no
+| The graph who the schema change job runs on. If not provided, means run the global schema change job.
```
@zhihuapengtg  `which graph to run the schema change job on, ...`

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+--
+====
+
+==== Parameters:
```
all parameters should be placed before Example

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+| `json`
+| no
+| get loading job information in json format
```
for all optional parameter, need to specify the default value, e.g.:
`default: false, don't get loading job detail information`

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+--
+[source,bash]
+----
+curl -H 'Content-Type: applicaiton/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/incidents_fraud_report_company_csv?graph=g&json=[true/false]' 
```
let's change `json` to `verbose`, need to change source code too

quanchentg committed:
fix message

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+| `gsql`
+| yes
+| Indicates whether to use the GSQL command for creation. Must be set to true. The request body should contain the GSQL command within the JSON object.
```
we don't support create vertex with json payload (not gsql command)?

dadongwang-tg replied:
it seems we supported in `create vertices (create vertices in global level)`, we should combine this two into one, and give two examples for each case

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+'''     
+
+=== create vertex (add global vertices to local graph)
```
change to "=== add global vertices to a local graph"

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+`POST /gsql/v1/schema/vertices`
+
+This endpoint is used to add existing global vertices to graph 
```
"to graph" --> "to a local graph"

romansod commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6394 @@
+
+=== get histogram statistics
+
+`GET /gsql/v1/stats/cardinality`
```
Should be 'histogram' here, and the example outputs are not really representative of the responses

romansod replied:
I was going to suggest to just link to my doc pages here but they seem to be lost (might be something wrong in the gsql-docs repo): https://github.com/tigergraph/gsql-docs/blob/02c6328cd6498500bea2f70ed5667dad3a0a83cc/modules/gsql-v2/querying/pages/query-optimizer/stats-api.adoc

Let me figure out from Victor what went wrong here, then we can make a link after

quanchentg committed:
fix saml/authnrequest response

ArthurXie-tg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

all parameters should be placed before Example

zhihuapengtg committed:
Update gsql-endpoints.adoc

quanchentg committed:
parameter position

ArthurXie-tg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

address comments

quanchentg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

all APIs should not have trailing /

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6054 @@
+
+| `verbose`
+| no
+| default: false, true means get loading job information in json format
```
this description is not correct. Please describe clearly when `verbose=false`, what will return, and `verbose=true` what will return

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6054 @@
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/abort?graph=g&jobIds=jobId1'
+curl -H 'Content-Type: application/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/abort?graph=g&jobIds=jobId1&isPause=true'
```
remove this line since `isPause=true` is default behavior

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6054 @@
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/jobId1?graph=g'
+curl -H 'Content-Type: application/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/jobId1?graph=g&isPause=true'
```
remove this line, since `isPause=true` is default behavior

quanchentg committed:
Update gsql-endpoints.adoc

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6053 @@
+=== create/update a loading job
+
+`POST /gsql/v1/loading-jobs`+
+`PUT /gsql/v1/loading-jobs`
```
lets split into two items, i.e.:
```
=== create a loading job
...
=== update an existing loading job
...

quanchentg committed:
Update gsql-endpoints.adoc

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6093 @@
+
+=== drop users
+
+`POST /gsql/scim/v2/Users/{id}`
```
this is wrong, should be `POST /gsql/scim/v2/Users`

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6093 @@
+--
+[source,bash]
+----
+curl -X POST -H "Content-type: application/json" "http://localhost:14240/gsql/scim/v2/Users/u1?gsqlFormat=false&action=delete"  -d ' {"userNames":["u1"]}'
```
remove `/u1?gsqlFormat=false`

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6093 @@
+--
+[source,bash]
+----
+curl -X PUT -H "Content-type: application/json" "http://localhost:14240/gsql/scim/v2/Users/u1?gsqlFormat=false"  -d '{"schemas":"[\"urn:ietf:params:scim:api:messages:2.0:PatchOp\"]","Operations":[{"op":"replace","path":"password","value":"newPassword"}]}'
```
remove `gsqlFormat=false` since the default is false, and similar for all others

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6093 @@
+
+| `queryName`
+| yes
+| the content of which query
```
`queryName` is not parameter, it is path variable

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6093 @@
+
+=== drop query
+
+`DELETE /gsql/v1/queires`
```
typo: `queires` -> `queries`

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6093 @@
+--
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X DELETE 'http://localhost:14240/gsql/v1/query=q1?graph=g'
```
the endpoint is wrong, should be:
/gsql/v1/queries?query=q1&graph=g

quanchentg committed:
Update gsql-endpoints.adoc

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6081 @@
+
+| `filter`
+| no
+| The filter format should follow this structure: `displayName op {value} op2 displayName op {value}`. Here, `op` can be one of `eq`, `nq`, `co`, `sw`, `ew`, and `op2` can be either `and` or `or`.
```
typo: `nq` -> `ne`

zhihuapengtg committed:
Update gsql-endpoints.adoc

add gsql/v1/users gsql/v1/groups API

zhihuapengtg committed:
update comments

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -4242,15 +4234,15 @@ Sample Request::
 [source,bash]
 ----
-curl -X GET "http://localhost:14240/gsql/scim/v2/Groups/g2?gsqlFormat=false"
+curl -X GET "http://localhost:14240/gsql//gsql/v1/groups/g2"
```
"/gsql//gsql" -> "/gsql"

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -4283,7 +4267,7 @@ Sample Request::
 [source,bash]
 ----
-curl -X GET "Content-type: text/plain" "http://localhost:14240/gsql/scim/v2/Groups?gsqlFormat=true"
+curl -X GET "Content-type: text/plain" "http://localhost:14240/gsql/scim/v2/Groups"
```
"/gsql/scim/v2/Groups" -> "/gsql/v1/groups"

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -4373,37 +4332,31 @@ Sample Request::
 ----
-curl -X POST "Content-type: application/json" "http://localhost:14240/gsql/scim/v2/Groups?gsqlFormat=false" -d '{"displayName":"scimG3","schemas":"[\"urn:ietf:params:scim:schemas:core:2.0:Group\"]","members":[{"display":"user3"},{"value":"user4"}]}'
+curl -X PATCH "Content-type: application/json" "http://localhost:14240/gsql/v1/groups/g1" -d '
+{"name": "g1", "rule": "rule1"}'
```
please give a valid rule, "rule1" is invalid

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -5023,24 +4990,62 @@ Sample Request::
-=== Create roles
 
-`POST /gsql/v1/roles`
+== SCIM APIS for users/groups
```
"APIS" -> "APIs"

zhihuapengtg committed:
address comments

zhihuapengtg committed:
Update gsql-endpoints.adoc

add statements api

zhihuapengtg committed:
Update gsql-endpoints.adoc

refactor v1/statements api description

zhihuapengtg committed:
Update gsql-endpoints.adoc

address comments:
change retrieve groups/users description,
change update  a group api path

zhihuapengtg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

arun-TG-PM commented on modules/API/pages/index.adoc:
```diff
@@ -0,0 +1,41 @@
+
+To submit a request, send an HTTP request to the NGINX server or the GSQL server.
+By default, the GSQL server listens for requests at port 8123 and the NGINX server listens on port 14240.
+A request needs to specify the following:
```
Let's rephrase to a much clearer sentence: "A request must include the following specifications:"

arun-TG-PM commented on modules/API/pages/index.adoc:
```diff
@@ -0,0 +1,41 @@
+* For `POST` requests, a data payload (i.e., request body) in JSON encoding unless otherwise specified
+* In some cases, request header parameters
+
+For requests that are sent to the GSQL server, the sender needs to provide TigerGraph user credentials for the request to be accepted.
```
Lets change it to "For requests sent to the GSQL server, TigerGraph user credentials must be provided for the request to be accepted"

arun-TG-PM commented on modules/API/pages/index.adoc:
```diff
@@ -0,0 +1,41 @@
+
+* `-d <data>`
+ ** Sends the specified data in a `POST` request to the HTTP server in the same way that a browser does when a user has filled in an HTML form and presses the submit button. This will cause curl to pass the data to the server using the content-type `application/x-www-form-urlencoded`.
+ ** If you start the data with the character `@`, the rest should be a filepath from which to read the data. The command `curl -d @foobar` will read data from a file named `foobar`.
```
Let's try to avoid "you" and "we".

Change it to: If the data starts with the character @, the rest should specify the filepath from which to read the data.

arun-TG-PM commented on modules/API/pages/index.adoc:
```diff
@@ -0,0 +1,41 @@
+ ** Makes curl fail silently (no output at all) on server errors.
+ ** This is mostly done to enable scripts etc. to better deal with failed attempts. In normal cases when an HTTP server fails to deliver a document, it returns an HTML document stating so (which often also describes why and more). This flag will prevent curl from outputting that and return error 22.
+* `-H <header>`
+ ** Extra header to include in the request when sending HTTP to a server. You may specify any number of extra headers.
```
Avoid "you" -> Any number of extra headers can be specified.

arun-TG-PM reviewed (commented):
Great work! Provided few comments and please resolve them soon.

ArthurXie-tg committed:
Update index.adoc

ArthurXie-tg commented:
@arun-TG-PM 
Thanks for reviewing! I just addressed your comments.

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

ArthurXie-tg committed:
Update gsql-endpoints.adoc

zhihuapengtg committed:
Update gsql-endpoints.adoc

dadongwang-tg commented on modules/API/pages/gsql-endpoints.adoc:
```diff
@@ -0,0 +1,6428 @@
+[source,bash]
+----
+curl -H 'Content-Type: application/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/abort?graph=financialGraph&jobIds=jobId1'
+curl -H 'Content-Type: application/json' -X GET 'http://localhost:14240/gsql/v1/loading-jobs/abort?graph=financialGraph&jobIds=jobId1&isPause=true'
```
remove this example, since `isPause=true` is internal use only
